apps/server/src/modules/common-cartridge/import/common-cartridge-file-parser.ts

```diff
@@ -0,0 +1,22 @@
+import AdmZip from 'adm-zip';
+import { CommonCartridgeManifestParser } from './common-cartridge-manifest-parser';
+
+export class CommonCartridgeFileParser {
+	public readonly manifest: CommonCartridgeManifestParser;
+
+	public constructor(file: Buffer) {
+		const archive = new AdmZip(file);
+
+		this.manifest = new CommonCartridgeManifestParser(this.getManifestFileAsString(archive));
+	}
+
+	private getManifestFileAsString(archive: AdmZip): string | never {
+		const manifest = archive.getEntry('imsmanifest.xml') ?? archive.getEntry('manifest.xml');
+
+		if (manifest) {
+			return archive.readAsText(manifest);
+		}
+
+		throw new Error('Manifest file not found');

```

Comment:
we could throw a specific error here with some more description that implements the loggable interface?

Reply:
Done.